pkg/sql/lease.go

```diff
@@ -785,22 +820,56 @@ func (t *tableState) acquireNodeLease(
 		return errors.New("cannot acquire lease when draining")
 	}
 
-	// Notify when lease has been acquired.
+	// Notify when lease is being acquired.
 	t.mu.acquiring = make(chan struct{})
+	t.mu.latestAcquisitionErr = nil
+
+	err := t.stopper.RunAsyncTask(
+		context.Background(), "sql.tableState: async table lease acquisition", func(ctx context.Context) {
```

Comment:
Done.

Although in the past experiences of having a long-running worker take in the context I'm a bit skeptical. But since it's a more short-lived task I'll give it a try :P.

In the case of a request spawning, a background refresh, passing through the current request context I think is not correct because the task can outlive the request. (But I'm also not quite sure what a context is or is for).

Reply:
(Have you read https://github.com/cockroachdb/cockroach/blob/master/docs/tech-notes/contexts.md?)

Reply:
> You can go a long way without understanding exactly what the point is (until you can’t), but that’s no way to live.
The story of my life right now. Thanks @benesch!